app/assets/javascripts/controllers/FileCtrl.coffee

```diff
@@ -43,37 +44,71 @@
     return
 
   ###*
-   * save file as bruse_file and append bruse_file to the file
-   * @param {obj} file  the file
+   * save file to temp array
+   * @param {obj} file  the file to add
   ###
   $scope.add = (file) ->
+    if file.is_dir and !file.contents
```

Comment:
This checks if the file is a folder, and if it is we want to add its contents. But if it doesn't have contents we have to get it. That's why both checks is present.

The next `else if` is if we already have loaded the contents of the folder.